packages/tests/src/_jest/helpers/env.ts

```diff
@@ -0,0 +1,135 @@
+    NEED_BUILD: boolean;
+    REQUESTED_BUNDLERS: string[];
+};
+
+export const getEnv = (argv: string[]): TestEnv => {
+    // Handle --cleanup flag.
+    const NO_CLEANUP = argv.includes('--cleanup=0');
+
+    // Handle --build flag.
+    const NEED_BUILD = argv.includes('--build=1');
+
+    // Handle --bundlers flag.
+    const REQUESTED_BUNDLERS = argv.includes('--bundlers')
+        ? argv[argv.indexOf('--bundlers') + 1].split(',')
+        : argv
+              .find((arg) => arg.startsWith('--bundlers='))
+              ?.split('=')[1]
+              .split(',') ?? [];
+
+    return {
+        NO_CLEANUP,
+        NEED_BUILD,
+        REQUESTED_BUNDLERS,
+    };
+};
+
+export const setupEnv = (env: TestEnv): void => {
+    const { NO_CLEANUP, NEED_BUILD, REQUESTED_BUNDLERS } = env;
+
+    if (NO_CLEANUP) {
+        process.env.NO_CLEANUP = '1';
+    }
+
+    if (NEED_BUILD) {
+        process.env.NEED_BUILD = '1';
+    }
+
+    if (REQUESTED_BUNDLERS.length) {
+        process.env.REQUESTED_BUNDLERS = REQUESTED_BUNDLERS.join(',');
+    }
+};
+
+export const logEnv = (env: TestEnv) => {
+    const { NO_CLEANUP, NEED_BUILD, REQUESTED_BUNDLERS } = env;
+    const envLogs = [];
+    if (NO_CLEANUP) {
+        envLogs.push(bgYellow(" Won't clean up "));
+    }
+
+    if (NEED_BUILD) {
+        envLogs.push(bgYellow(' Will also build used plugins '));
+    }
+
+    if (REQUESTED_BUNDLERS.length) {
+        if (
+            !(REQUESTED_BUNDLERS as BundlerFullName[]).every((bundler) =>
+                FULL_NAME_BUNDLERS.includes(bundler),
+            )
+        ) {
+            throw new Error(
+                `Invalid "${red(`--bundlers ${REQUESTED_BUNDLERS.join(',')}`)}".\nValid bundlers are ${FULL_NAME_BUNDLERS.map(
+                    (b) => green(b),
+                )
+                    .sort()
+                    .join(', ')}.`,
+            );
+        }
+        const bundlersList = REQUESTED_BUNDLERS.map((bundler) => green(bundler)).join(', ');
+        envLogs.push(`Running ${bgYellow(' ONLY ')} for ${bundlersList}.`);
+    }
+
+    if (!NO_CLEANUP || !NEED_BUILD || REQUESTED_BUNDLERS.length) {
+        const tips: string[] = [];
+        if (!NO_CLEANUP) {
+            tips.push(`  ${green('--cleanup=0')} to keep the built artifacts.`);
+        }
+        if (!NEED_BUILD) {
+            tips.push(`  ${green('--build=1')} to force the build of the used plugins.`);
+        }
+        if (!REQUESTED_BUNDLERS.length) {
+            tips.push(`  ${green('--bundlers=webpack4,esbuild')} to only use specified bundlers.`);
+        }
+        envLogs.push(dim(`\nYou can also use : \n${tips.join('\n')}\n`));
+    }
+
+    if (envLogs.length) {
+        console.log(`\n${envLogs.join('\n')}\n`);
+    }
+};
+
+export const getOutDir = (workingDir: string, folderName: string): string => {
+    return path.resolve(workingDir, `./dist/${folderName}`);
+};
+
+const FIXTURE_DIR = path.resolve(__dirname, '../fixtures');
```

Comment:
Not a blocker, but `__dirname` doesn't exist in ESM, only in CJS. Idk if this can create issues or not

Reply:
I'll update once we transition to esm.